pymeasure/instruments/proterial/rod4.py

```diff
@@ -0,0 +1,144 @@
+        validator=strict_discrete_set,
+        values={'flow': 0, 'close': 1, 'open': 2},
+        map_values=True
+    )
+
+    def __init__(self, *argv, **kwargs):
+        super().__init__(*argv, **kwargs)
+        self._flow_unit = None
+
+    @property
+    def flow_unit(self):
+        """Control flow units for the selected channel.
+        Valid options are %, sccm, or slm.
+        Display in absolute units is in sccm for control range < 10 slm."""
+        if self._flow_unit is None:
+            log.info("Flow units have not been set on ROD-4 channel "
+                     f"{self.id}")
+        return self._flow_unit
+
+    @flow_unit.setter
+    def flow_unit(self, units):
+        values = {'%': 0, 'sccm': 1, 'slm': 1}
+        if isinstance(units, str) and units.casefold() in values:
+            units = units.casefold()
+            self.write("\x020{ch}SFU%d" % values[units])
+            self._flow_unit = 'sccm'
+        else:
+            log.info('Units must be %, sccm, or slm.')
+
+
+class ROD4(Instrument):
+    """Represents the Proterial ROD-4(A) operator for mass flow controllers
+    and provides a high-level interface for interacting with the instrument.
+    User must specify which channel to control (1-4).
+
+    .. code-block:: python
+
+        rod4 = ROD4("ASRL1::INSTR")
+
+        print(rod4.version)             # Print version and series number
+        rod4.ch_1.mfc_range = 500       # Sets Channel 1 MFC range to 500 sccm
+        rod4.ch_2.valve_mode = 'flow'   # Sets Channel 2 MFC to flow control
+        rod4.ch_3.setpoint = 50         # Sets Channel 3 MFC to flow at 50% of full range
+        print(rod4.ch_4.actual_flow)    # Prints Channel 4 actual MFC flow in %
+
+    """
+    ch_1 = Instrument.ChannelCreator(ROD4Channel, 1)
+    ch_2 = Instrument.ChannelCreator(ROD4Channel, 2)
+    ch_3 = Instrument.ChannelCreator(ROD4Channel, 3)
+    ch_4 = Instrument.ChannelCreator(ROD4Channel, 4)
+
+    def __init__(self, adapter, name="ROD-4 MFC Controller", **kwargs):
+        super().__init__(
+            adapter, name, read_termination='\r', write_termination='\r', **kwargs
+        )
+        self._keyboard = None
+
+    version = Instrument.measurement(
+        "\x0200RVN",
+        """Get the version and series number. Returns x.xx<TAB>S/N """
+    )
+
+    @property
+    def keyboard(self):
+        """Control the front keyboard lock status.
+        Valid options are unlocked or locked. """
+        if self._keyboard is None:
+            log.info('ROD-4 keyboard status has not been set')
+        return self._keyboard
+
+    @keyboard.setter
+    def keyboard(self, status):
+        values = {'unlocked': 0, 'locked': 1}
+        if isinstance(status, str) and status.casefold() in values:
+            status = status.casefold()
+            self.write("\x0200SKO%d" % values[status])
+            self._keyboard = values[status]
+        else:
+            log.info('Status must be unlocked or locked.')
```

Comment:
See above for stored state of instrument.

Reply:
Changed to a setting property